### PR TITLE
Run rule2

### PR DESCRIPTION
**Pull request by rkoshak**
Address #77 . Second attempt at this PR to fix DCO errors. Replaces PR #81.

### CODE REVIEW

**Review by florian-h05 — CHANGES_REQUESTED** [review]
Two suggestions for the JSDoc comment of runRule.
Please have a look at my comments.

Apart from that it looks good to me :-).

**Comment by florian-h05** [comment]
@rkoshak: May it helps you

When you use VS Code and you have installed the git extension, you can enable `Always Sign Off` under `Settings` -> `Extensions` -> `Git`.

And with the sidebar‘s Git tab, you can revert the last commit by pressing the three points (in the upper right corner of that tab) -> Commit.
In the filed for the commit message, you can use arrow up/down to reuse a commit message. After committing a reverted commit, you MUST NOT press the sychronize button, instead open the Terminal inside VS Code and force-push (e.g. `git push origin --force`).

File: rules/rules.js [hunk 1]
```diff
@@ -108,6 +108,56 @@ const removeRule = function (uid) {
...
 };
 
+
+/**
+ * Runs the rule with the given UID. Throws errors when the rule doesn't exist
+ * or is unable to run (e.g. it's disabled).
+ * 
+ * @memberOf rules
+ * @param {String} uid the UID of the rule to run
+ * @param {Map<Object>} args optional dict of data to pass to the called rule
```
[the remark this answers is not in the document]
Comment (rkoshak) [reply rules/rules.js]: Done

File: rules/rules.js [hunk 2]
```diff
@@ -108,6 +108,56 @@ const removeRule = function (uid) {
...
+ * @param {Boolean} cond when true, the called rule will only run if it's conditions are met
```
[the remark this answers is not in the document]
Comment (rkoshak) [reply rules/rules.js]: Done

**Comment by rkoshak** [comment]
Thanks for the tips! I've changed the setting so hopefully won't run into this problem again, as long as I'm using VSCode at least.

**Review by florian-h05 — APPROVED** [review]
LGTM!

For anyone watching: I have tested the changes.

**Comment by Confectrician** [comment]
Just as a sidenote @rkoshak

If you are using vscode you can enable signing off comits as default, evenb if you are working with the git sidebar in vscode.

THe setting is
```
"git.alwaysSignOff": true
```

You just have to configure the proper sign off credentials in git.
(Which you probably have done already for a manual sign off via console commit already anyway.)

File: rules/rules.js [hunk 3]
```diff
@@ -87,7 +87,7 @@ const getGroupsForItem = function (ruleConfig) {
  * @returns {Boolean} whether the rule exists
  */
 const ruleExists = function (uid) {
-    return !(ruleRegistry.get(uid) == null);
+    return !(RuleManager.getStatusInfo(uid) == null);
```
Comment (digitaldan) [inline rules/rules.js:90]: Out of curiosity, whats the benefit of changing this line ?

**Comment by digitaldan** [comment]
Thanks @rkoshak this looks good, i only have one comment.  And thanks @florian-h05 for testing.

File: rules/rules.js [hunk 4]
```diff
@@ -108,6 +108,56 @@ const removeRule = function (uid) {
...
 
+
+/**
+ * Runs the rule with the given UID. Throws errors when the rule doesn't exist
+ * or is unable to run (e.g. it's disabled).
+ * 
+ * @memberOf rules
+ * @param {String} uid the UID of the rule to run
+ * @param {Map<Object>} [args={}] args optional dict of data to pass to the called rule
+ * @param {Boolean} [cond=true] when true, the called rule will only run if it's conditions are met
```
Comment (digitaldan) [inline rules/rules.js:119]: should we add a throws line as well?  for example
```javascript
@throws Will throw an error if the rule is not found or is disabled
```

and also in the two functions below ? 

https://jsdoc.app/tags-throws.html

File: rules/rules.js (hunk 3 above)
Comment (rkoshak) [reply rules/rules.js:90]: I think I lost that comment. In UI rules, using the ruleRegistry doesn't work for some reason after the rule is run the second time around. The RuleRegistry gets stale somehow. But the RuleManager works every time.

File: rules/rules.js (hunk 4 above)
Comment (rkoshak) [reply rules/rules.js:119]: I started to but it's not on any of the other methods that throw Errors in this file so I stuck with that as the convention. Should I add @throws to them all or just mine?

Comment (digitaldan) [reply rules/rules.js:119]: Thanks!

**Review by digitaldan — APPROVED** [review]
LGTM

**Pull request merged**